src/PlayersManager.ts

```diff
@@ -712,17 +712,16 @@ export class PlayersManager extends Dispatcher {
         const captionsTracks = tracks.find(
           (track) => track._kind === "subtitles"
         );
-        if (
-          captionsTracks ||
-          (audioTracks &&
-            audioTracks._language &&
-            audioTracks._language != undefined)
-        ) {
-          // this video has captions or audio
+        if (audioTracks && audioTracks._language && audioTracks._language != undefined) {
           const view = player.getView().parentElement.parentElement
             .parentElement;
           view.classList.add("has-extra-tracks");
         }
+        if (captionsTracks){
+          const view = player.getView().parentElement.parentElement
+              .parentElement;
+          view.classList.add("has-captions");
```

Comment:
does the class never need to be removed ?

Reply:
no, it's relevant as long as this video is played, when you move to another one player will create again with the relevant classes.